.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,33 @@
+name: Pre-commit Checks
+
+on: 
+  pull_request:
+    paths:
+      - ".github/workflows/pre-commit.yml"
+      - ".pre-commit-config.yaml"
+      - "tuning/*"
```

Comment:
I just found out that these should be `tuning/**` (double asterisk). Oops >_>

I also found out that path filters are incompatible with "required checks". More reason to get this code moved into its own repo so the checks can just always run since every PR in the repo will want to run them and ensure they stay passing.